packages/admin-ui-docs/docs/guidelines/components/button.mdx

```diff
@@ -1,131 +1,268 @@
 ---
 title: Button
 path: /button/
+hide_table_of_contents: true
 ---
 
-Buttons trigger an action, or allow the user to advance a state. They are where the user interacts with the screen, enabling actions or decisions.
-
-## Usage
+import { Page, Panel } from '@site/src/components/Page'
+import { PageSection } from '@site/src/components/PageSection'
+import { CodeSection } from '@site/src/components/CodeSection'
+import { MultipleCodeSection } from '@site/src/components/MultipleCodeSection'
 
-```jsx isStatic
-import { Button } from '@vtex/admin-ui'
-
-export function Example() {
-  return <Button onClick={() => {}}>Admin UI Button</Button>
+export const tabToc = {
+  overview: [
+    {
+      id: 'variants',
+      value: 'Variants',
+      level: 2,
+    },
+    {
+      id: 'how-does-each-variant-work',
+      value: 'How does each variant work?',
+      level: 3,
+    },
+    {
+      id: 'button-sizes',
+      value: 'Button sizes',
+      level: 2,
+    },
+    {
+      id: 'button-with-icon',
+      value: 'Button with icon',
+      level: 2,
+    },
+    {
+      id: 'disabled',
+      value: 'Disabled',
+      level: 2,
+    },
+    {
+      id: 'loading',
+      value: 'Loading',
+      level: 2,
+    },
+  ],
+  codeReference: [
+    {
+      id: 'usage',
+      value: 'Usage',
+      level: 2,
+    },
+    {
+      id: 'props',
+      value: 'Props',
+      level: 2,
+    },
+  ],
 }
-```
 
-## Examples
+<Page toc={tabToc} hasTabs>
+<Panel id="overview">
 
-### Variant
+Buttons trigger an action, or allow the user to advance a state. They are where the user interacts with the screen, enabling actions or decisions.
 
-The variant defines the appearance and emphasis of the button.
+<MultipleCodeSection
+  isLive
+  id={tabToc.overview[0].id}
+  title={tabToc.overview[0].value}
+  layout="grid"
+  optionsTitle="Variant"
+  options={[
+    {
+      id: 'primary',
+      label: 'Primary',
+      columnsCount: 2,
+      code: `<>
+  <div className="grid-item">
```

Comment:
This is getting super complex i think. Any alternative? I do not see non-developers updating docs with this structure.

Reply:
To be honest, to be able to update this documentation with this level of design customization a new level o complexity is added.